### PR TITLE
Feat/es6

### PR DESCRIPTION
**Pull request by pmast**

### CODE REVIEW

File: src/array-stream.js [hunk 1]
```diff
@@ -1,29 +1,28 @@
-const stream = require('stream'),
-  util = require('util');
+const stream = require('stream');
```
Comment (ro-ka) [inline src/array-stream.js]: Why not import?

File: src/array-stream.js [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
   }
 
-  this.push(this.values[this.currentIndex++]);
-};
+  /**
+   * The _read function for the addresses stream.
+   */
+  /* eslint-disable no-underscore-dangle */
+  _read() {
+  /* eslint-enable no-underscore-dangle */
```
Comment (ro-ka) [inline src/array-stream.js]: Use `// eslint-disable-line no-underscore-dangle` --> shorter.

File: src/cache.js [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
-  address,
-  result,
-  callback = function() {}
-) {
-  this.db.put(address, result, (error) => {
-    if (error) {
-      throw error;
-    }
+export default class Cache {
+  constructor(cacheFile = 'geocache.db') {
```
Comment (ro-ka) [inline src/cache.js:14]: DocBlock!

File: src/cache.js [hunk 4]
```diff
@@ -1,43 +1,38 @@
...
-};
+      callback();
+    });
+  }
 
-module.exports = Cache;
+  /**
+   * Add new entries to the Cache
+   * @param {String} address  The address that shall be cached
+   * @return {Object} The geocoded result
```
Comment (ro-ka) [inline src/cache.js]: Whoops, the descriptions were wrong… :astonished:

File: src/geocode-stream.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  /* eslint-disable no-underscore-dangle */
+  /**
+   * The _transform function for the stream.
+   * @param {String}   address The address to geocode
+   * @param {String}   encoding The encoding
+   * @param {Function} done The done callback function
+   */
+  _transform(address, encoding, done) {
```
Comment (ro-ka) [inline src/geocode-stream.js]: Disable only the line if possible.

File: src/geocoder.js [hunk 6]
```diff
@@ -8,90 +8,91 @@ const Cache = require('./cache'),
...
- * Start geocoding a single address
- * @param {String} address The address to geocode
- * @param {Function} resolve The Promise resolve function
- * @param {Function} reject The Promise reject function
- * @return {?} Something to get out
- */
-Geocoder.prototype.startGeocode = function(address, resolve, reject) {
-  const cachedAddress = this.cache.get(address);
+    this.cache = new Cache(options.cacheFile);
+    this.googlemaps = require('googlemaps');
```
Comment (ro-ka) [inline src/geocoder.js]: Use import at the top.

File: src/index.js [hunk 7]
```diff
@@ -1,103 +1,39 @@
...
-
-      data.error = error.message;
-      this.push(data);
-      done();
+export default class GeoBatch {
+  constructor(options = {}) {
+    this.geocoder = new Geocoder({
+      cacheFile: options.cacheFile || 'geocache.db',
+      clientId: options.clientId,
+      privateKey: options.privateKey
```
Comment (ro-ka) [inline src/index.js]: Use destructing and a default value for the params.

**Comment by ro-ka** [comment]
:checkered_flag:

File: src/cache.js (hunk 3 above)
Comment (pmast) [inline src/cache.js:14]: Is there but unchanged. Will adjusted the type in the DocBlock though.

Comment (ro-ka) [inline src/cache.js:14]: But for the constructor. :)

**Comment by pmast** [comment]
:cactus:

**Pull request merged**